### PR TITLE
LPD-16334 Unauthorized access to workflow definition via API

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-16334](https://liferay.atlassian.net/browse/LPD-16334)

### CODE REVIEW

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-integration-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/integration/internal/WorkflowDefinitionManagerImpl.java [hunk 1]
```diff
@@ -109,12 +112,26 @@ public List<WorkflowDefinition> getActiveWorkflowDefinitions(
...
+						serviceContext);
+			}
+			else {
+				kaleoDefinitions =
+					_kaleoDefinitionService.getScopeKaleoDefinitions(
+						WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
+						KaleoDefinitionOrderByComparator.getOrderByComparator(
+							orderByComparator, _kaleoWorkflowModelConverter),
+						serviceContext);
+			}
```
Comment (MarinhoFeliphe) [inline modules/apps/portal-workflow/portal-workflow-kaleo-runtime-integration-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/integration/internal/WorkflowDefinitionManagerImpl.java:134]: Create a private method in this class that will remove the else statement and simplify the code, for example:

````java
private <T> T _get(
    UnsafeSupplier<T> localServiceSupplier,
    UnsafeSupplier<T> serviceSupplier) {

    return KaleoDefinitionResourcePermissionThreadLocal.isDisabled() ? 
                localServiceSupplier.get() : serviceSupplier.get();
}
````

so

````java
List<KaleoDefinition> kaleoDefinitions = _get(
    () -> _kaleoDefinitionLocalService.getScopeKaleoDefinitions(
		WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
		KaleoDefinitionOrderByComparator.getOrderByComparator(
			orderByComparator, _kaleoWorkflowModelConverter), serviceContext),
    () -> _kaleoDefinitionService.getScopeKaleoDefinitions(
		WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
		KaleoDefinitionOrderByComparator.getOrderByComparator(
			orderByComparator, _kaleoWorkflowModelConverter), serviceContext));
````

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:workflow

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Pull request closed without merging**